core/trino-spi/src/main/java/io/trino/spi/connector/RecordPageSource.java

```diff
@@ -29,18 +31,25 @@
     private static final int ROWS_PER_REQUEST = 4096;
     private final RecordCursor cursor;
     private final List<Type> types;
+    private final Metrics metrics;
```

Comment:
Instead of sending dummy metrics via MockConnectorFactory, you can consider defining some actual metric like `rowCount` in MockConnectorPageSource if we get non-empty pages there during tests.